clientd3d/d3drender.c

```diff
@@ -7012,8 +7016,14 @@ void D3DRenderObjectsDraw(d3d_render_pool_new *pPool, room_type *room,
 		// Nodes with a bound height adjust are part of other players' upper bodies.
 		if (pRNode->boundingHeightAdjust == 0)
 		{
-			// Typical items such as reagents, keys, etc.
-			pChunk->zBias = ZBIAS_DEFAULT + (z_depth_inc++);
+			// Typical items such as reagents, keys, etc. are drawn at the default depth
+			// offset by the number of items already drawn at this location.
+
+			// Combine objects x and y position into a single int64 for the map key.
+			int64_t key = ((int64_t)pRNode->motion.x << 32) | (int32_t)(pRNode->motion.y & 0xFFFFFFFF);

```

Comment:
Since there's no longer any binning, this will only affect objects precisely on top of each other.  That will take care of piles of objects dropped simultaneously, which might be enough to solve the problem, just checking that it's what you want.